server/__test__/user.spec.js

```diff
@@ -154,4 +155,64 @@ describe('User tests', () => {
         });
     });
   });
+
+  describe('test for email signin', () => {
+    it('Should sign in a user with complete form data', async done => {
+      server()
+        .post(`${url}/auth/signin`)
+        .send(mockUser.completeLoginData)
+        .end((err, res) => {
+          expect(res.statusCode).toEqual(200);
+          expect(res.body.message).toEqual('Login successful');
+          expect(res.body.status).toEqual(200);
+          expect(res.body.data).toHaveProperty('token');
+          expect(res.body.data).toHaveProperty('id');
+          expect(res.body.data).toHaveProperty('firstName');
+          expect(res.body.data).toHaveProperty('lastName');
+          expect(res.body.data).toHaveProperty('email');
+          done();
+          expect(Object.keys(res.body.data)).toMatchSnapshot();
+        });
+    });
+
+    it('Should not sign in with incomplete form data', async done => {
+      server()
+        .post(`${url}/auth/signin`)
+        .send(mockUser.incompleteLoginData)
+        .end((err, res) => {
+          expect(res.statusCode).toEqual(400);
+          expect(res.body).toHaveProperty('error');
+          expect(res.body.error[0]).toEqual('Email should not be left empty: Please input email address');
+          expect(res.body.error[1]).toEqual('Password should not be empty: Please input password');
+          done();
+          expect(res.body).toMatchSnapshot();
+        });
+    });
+
+    it('Should not sign in user with wrong email', async done => {
+      server()
+        .post(`${url}/auth/signin`)
+        .send(mockUser.wrongEmail)
+        .end((err, res) => {
+          expect(res.statusCode).toEqual(404);
```

Comment:
Bro here's what I think. It should be a 401 error. The user is not searching for a resource, he's trying to log in. So it should be a 401 if the email and password actually match the correct format, and if they don't match the format for instance `wrong-email@mail` as opposed to `wrong-email@mail.com`, it should be a 400 error.

Reply:
me and these status codes. 
Thanks bro